paasta_tools/deployd/master.py

```diff
@@ -136,15 +136,39 @@ def main_loop(self):
                 message = None
             if message == "ABORT":
                 break
+            if not self.all_watchers_running():
+                self.log.error("One or more watcher died, committing suicide!")
+                break
+            if self.all_workers_dead():
+                self.log.error("All workers have died, comitting suicide!")
+                break
```

Comment:
It'd be clearer if they did! It doesn't matter to the init system of course but I agree this should be `sys.exit(1)`